IncludeToolbox/IncludeWhatYouUse/IWYU.cs

```diff
@@ -263,64 +256,69 @@ static public string RunIncludeWhatYouUse(string fullFileName, EnvDTE.Project pr
-                    process.StartInfo.Arguments += "-Xiwyu --max_line_length=1024 ";
-
-                    // Custom stuff.
-                    process.StartInfo.Arguments += settings.AdditionalParameters;
-                    process.StartInfo.Arguments += " ";
+                    settings.MappingFiles = new string[] { "stl.c.headers.imp", "msvc.imp", "boost-all.imp", "boost-all-private.imp" };
+                }
+                for (int i = 0; i < settings.MappingFiles.Length; ++i)
+                    iwyuOptions += "-Xiwyu --mapping_file=\"" + settings.MappingFiles[i] + "\" ";
+                if (settings.NoDefaultMappings)
+                    iwyuOptions += "-Xiwyu --no_default_mappings ";
+                if (settings.PCHInCode)
+                    iwyuOptions += "-Xiwyu --pch_in_code ";
+                switch (settings.PrefixHeaderIncludes)
+                {
+                    case IncludeWhatYouUseOptionsPage.PrefixHeaderMode.Add:
+                        iwyuOptions += "-Xiwyu --prefix_header_includes=add ";
+                        break;
+                    case IncludeWhatYouUseOptionsPage.PrefixHeaderMode.Remove:
+                        iwyuOptions += "-Xiwyu --prefix_header_includes=remove ";
+                        break;
+                    case IncludeWhatYouUseOptionsPage.PrefixHeaderMode.Keep:
+                        iwyuOptions += "-Xiwyu --prefix_header_includes=keep ";
+                        break;
                 }
+                if (settings.TransitiveIncludesOnly)
+                    iwyuOptions += "-Xiwyu --transitive_includes_only ";
+
+                // Set max line length so something large so we don't loose comment information.
+                // Documentation:
+                // --max_line_length: maximum line length for includes. Note that this only affects comments and alignment thereof,
+                // the maximum line length can still be exceeded with long file names(default: 80).
+                iwyuOptions += "-Xiwyu --max_line_length=1024 ";
+
+                // Include-paths and Preprocessor.
+                var includeEntries = VSUtils.GetProjectIncludeDirectories(project, false);
+                var includes = includeEntries.Aggregate("", (current, inc) => current + ("-I \"" + Regex.Escape(inc) + "\" "));
 
+                var defines = " -D " + string.Join(" -D", preprocessorDefintions.Split(';'));
 
-                // Finally, the file itself.
-                process.StartInfo.Arguments += "\"";
-                process.StartInfo.Arguments += fullFileName;
-                process.StartInfo.Arguments += "\"";
+                // write support file. Long argument lists lead to an error. Support files are the solution here.
+                // https://github.com/Wumpf/IncludeToolbox/issues/36
+                var supportFile = Path.GetTempFileName();
+                var supportContent = includes + " " + defines + " " + Regex.Escape(fullFileName);                
```

Comment:
I don't think Regex.Escape is what you want here. As I understand it this is for escaping regex metacharacters so they are matched as characters. but fullFileName is clearly not a regex. So just put it in quotes as it was before.

Reply:
You are right. I will have a look on this soon and fix the PR.